### DISCUSSION TITLE
Getting it running in Google Colab

### DISCUSSION THREAD

**Discussion by franzenzenhofer** [joshcarty/google-searchconsole#54] [Q&A] [answered]
hi, i am trying to get it to run in Google Colab 

https://colab.research.google.com/drive/1lpGPtbYGVgmPhd8QwGciQY5zl1lP7MLo#scrollTo=2gM4riuS5TQg 

but even on a complete new environment I get this error

```
`---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
[<ipython-input-3-4a99930e4989>](https://localhost:8080/#) in <module>()
----> 1 account = searchconsole.authenticate(client_config='client_secrets.json')

6 frames
[/usr/lib/python3.7/socketserver.py](https://localhost:8080/#) in server_bind(self)
    464         if self.allow_reuse_address:
    465             self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
--> 466         self.socket.bind(self.server_address)
    467         self.server_address = self.socket.getsockname()
    468 

OSError: [Errno 98] Address already in use`
```

is there an Google Colab example where somebody got it running? Googling did not help. Thx

**Comment by joshcarty** [comment, marked as answer]
Hi @franzenzenhofer,

From Google Colab, you'll want to do:

```python
account = searchconsole.authenticate(client_config='auth/client_secrets.json', flow='console')
```

For more context:

By default, the package tries to help you authenticate by opening a local web server and browser for you to sign in and authenticate with. This is the "web" flow. It's not possible to do that on Google Colab so there is an alternative "console" flow which prints a link that you can manually open, authenticate with, and copy and paste into your notebook.

**Reply by Swchuynh** [reply]
Ppi phelsn.

**Reply by Swchuynh** [reply]
No.! Sale

**Reply by TedTansley** [reply]
Awesome, this worked for me quite well. Had been struggling for some time getting Search Console's API to work and this made it so much easier. 
I ended up changing the client_config=to the json path in colab. Will look to make that easier on myself in the future. Thanks for this!
